homeassistant/components/rituals_perfume_genie/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Config flow for Rituals Perfume Genie integration."""
+import logging
+
+from aiohttp import ClientResponseError
+from pyrituals import Account, AuthenticationException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import ACCOUNT_HASH, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Rituals Perfume Genie."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+        errors = {}
+
+        await self.async_set_unique_id(user_input[CONF_EMAIL])
+        self._abort_if_unique_id_configured()
+
+        try:
+            session = async_get_clientsession(self.hass)
+            account = Account(user_input[CONF_EMAIL], user_input[CONF_PASSWORD])
+            await account.authenticate(session)
+        except ClientResponseError:
+            errors["base"] = "cannot_connect"
+        except AuthenticationException:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
```

Comment:
```
# pytest --cov=homeassistant/components/rituals_perfume_genie/ --cov-report term-missing -- tests/components/rituals_perfume_genie/test_*.py
Test session starts (platform: linux, Python 3.8.1, pytest 6.2.2, pytest-sugar 0.9.4)
rootdir: /root/home-assistant, configfile: pyproject.toml
plugins: aiohttp-0.3.0, profiling-1.7.0, parallel-0.1.0, anyio-1.3.0, forked-1.1.3, mock-3.1.1, socket-0.3.5, asyncio-0.14.0, aresponses-2.0.0, test-groups-1.0.3, requests-mock-1.8.0, reportlog-0.1.2, rerunfailures-9.1.1, cov-2.10.1, sugar-0.9.4, timeout-1.4.2, xdist-2.1.0
collecting ... 
 tests/components/rituals_perfume_genie/test_config_flow.py ✓✓                                                                                                      100% ██████████

----------- coverage: platform linux, python 3.8.1-final-0 -----------
Name                                                            Stmts   Miss  Cover   Missing
---------------------------------------------------------------------------------------------
homeassistant/components/rituals_perfume_genie/config_flow.py      32      4    88%   45, 48-50


Results (1.78s):
       2 passed
```

Please add coverage for missing lines

Reply:
I have added two tests ([f8e61d1](https://github.com/home-assistant/core/pull/46218/commits/f8e61d111e6eee24b2596b3f084f745a43606893)). The coverage is now 100%.
I was not aware that we had to have 100% coverage, must have missed it in the documentation. Thank you for pointing that out.